resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/submit/ClientSuite.scala

```diff
@@ -177,13 +176,13 @@ class ClientSuite extends SparkFunSuite with BeforeAndAfter {
   }
 
   test("Waiting for app completion should stall on the watcher") {
+    kconf.sparkConf.set(WAIT_FOR_APP_COMPLETION, true)
```

Comment:
This isn't technically needed, since `watchOrStop` will be called regardless, right?

Reply:
I guess this is needed because it won't stall when false

Reply:
Sorry, not following. The only code that "stalls" is `LoggingPodStatusWatcherImpl`, and this test is using a mock, not that implementation. So what exactly is using this config?

BTW that question applies to all changes you're making in this class.

Reply:
I see, thanks